qualtran/bloqs/classical_quantum_conversions.py

```diff
@@ -0,0 +1,180 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+from typing import Dict
+
+from attrs import frozen
+
+from qualtran import (
+    Bloq,
+    BloqBuilder,
+    QBit,
+    QInt,
+    QIntOnesComp,
+    QUInt,
+    Register,
+    Signature,
+    SoquetT,
+)
+from qualtran.bloqs.basic_gates import XGate
+from qualtran.cirq_interop.bit_tools import iter_bits, iter_bits_twos_complement
+from qualtran.simulation.classical_sim import ClassicalValT
+
+
+@frozen
+class ClassicalToQBit(Bloq):
+    r"""A 1-bit bloq which converts a classical bit of information into a QBit of information.
+        Assumes the input register is initialized to 0.
+    Args:
+        k: A classical bit value to be represented as a QBit.
+    Registers:
+        x: A 1-bit input register.
+    """
+
+    k: int
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        return Signature([Register('x', QBit())])
+
+    def on_classical_vals(self, x: 'ClassicalValT') -> Dict[str, 'ClassicalValT']:
+        assert (
+            self.k == 0 or self.k == 1
+        ), "Argument `k` should be a classical bit containing value 0 or 1"
```

Comment:
this sort of validation probably should be done at construction time. You can use attrs validators

```python
k: int = field(validator=lambda k: k in [0, 1])
```

or similar